features/weth/Weth.tsx

```diff
@@ -84,27 +86,43 @@ const Weth = () => {
   const { signer } = Connection.useContainer();
   const { getEtherscanUrl } = Etherscan.useContainer();
 
-  const [ethAmount, setEthAmount] = useState<BigNumberish | null>(null);
-  const [wethAmount, setWethAmount] = useState<BigNumberish | null>(null);
+  const [ethAmount, setEthAmount] = useState<string>("0");
+  const [wethAmount, setWethAmount] = useState<string>("0");
   const [hash, setHash] = useState<string | null>(null);
   const [success, setSuccess] = useState<boolean | null>(null);
   const [error, setError] = useState<Error | null>(null);
 
-  const _submitTxn = async (amount: string | null, action: ACTION_TYPE) => {
-    if (weth) {
-      if (amount && utils.parseEther(amount).gt(0)) {
+  if (
+    weth !== null &&
+    wethBalance !== null &&
+    ethBalance !== null &&
+    signer !== null
+  ) {
```

Comment:
nit: thoughts on just doing an early return if these are null? Then all the happy path logic wouldn't be nested and this might be slightly easier to follow.

Something like:
```
if (weth === null && ... ) {
  // Return loading component
}

// More complicated happy path logic
```

Reply:
There is an early return in the `else` component of this if statement, its just buried at the bottom of the file. I could technically reverse the boolean check to `if(condition === null || condition === null)...` which would have the early return show up earlier in the file but this is how I set it up with rest of repo